### PR TITLE
modules: Add eSPI SAF header and updates

### PR DESCRIPTION
**Pull request by scottwcpg**
Add Microchip MEC15xx eSPI SAF header and SAF changes
in other affected hardware headers: espi_io and qmspi.

Signed-off-by: Scott Worley <scott.worley@microchip.com>

### CODE REVIEW

**Comment by scottwcpg** [comment]
Need review from @albertofloyd, @franciscomunoz

**Comment by albertofloyd** [comment]
@scottwcpg after accepting the invite, I was able to add myself as reviewer. 
Check if you can add other people yourself?

**Comment by scottwcpg** [comment]
Hi Alberto,

I see your name under “Reviewer” but there is no pull down list per GitHub documentation.  I can’t add anyone to https://github.com/zephyrproject-rtos/hal_microchip

In the review section.

**Comment by franciscomunoz** [comment]
Hi @scottwcpg I just have a question related to  versioning. Where can we see the version bump for these HAL changes?

File: mec/mec1501/component/qmspi.h [hunk 1]
```diff
@@ -143,14 +145,14 @@
...
-#define MCHP_QMSPI_M_SIG_MODE0		(0x00ul << MCHP_QMSPI_M_SIG_POS)
+#define MCHP_QMSPI_M_SIG_MODE0		0x00ul
 #define MCHP_QMSPI_M_SIG_MODE1		(0x06ul << MCHP_QMSPI_M_SIG_POS)
 #define MCHP_QMSPI_M_SIG_MODE2		(0x01ul << MCHP_QMSPI_M_SIG_POS)
 #define MCHP_QMSPI_M_SIG_MODE3		(0x07ul << MCHP_QMSPI_M_SIG_POS)
 #define MCHP_QMSPI_M_CS_POS		12u
 #define MCHP_QMSPI_M_CS_MASK0		0x03ul
 #define MCHP_QMSPI_M_CS_MASK		(0x03ul << 12)
-#define MCHP_QMSPI_M_CS0		(0x00ul << 12)
+#define MCHP_QMSPI_M_CS0		0x00ul
```
Comment (albertofloyd) [inline mec/mec1501/component/qmspi.h:155]: @scottwcpg could you confirm this is the latest value, in the original espi_qmspi_update branch shared with SAF driver this was 
#define MCHP_QMSPI_M_CS0		(0x00ul << 12)

**Comment by scottwcpg** [comment]
> Hi @scottwcpg I just have a question related to versioning. Where can we see the version bump for these HAL changes?
> 
> Can you add this information in the commit message?
> Also the origin Repo
> 
> ```
> Origin:
>     MCHP
>         https://github.com/XXXXX/XXXX
> 
> Status:
>     version X.X.X
> ```

Hi @franciscomunoz 
I will add versioning as requested.
Please wait for a rebase.

File: mec/mec1501/component/qmspi.h (hunk 1 above)
Comment (scottwcpg) [reply mec/mec1501/component/qmspi.h:155]: Hi @albertofloyd 
I was trying out one of the linting tools in neovim and it complained that shifting the value 0 was was redundant.
I will put it back for clarity in next rebase.

**Pull request merged**